### PR TITLE
Configurable build resources in Function UI

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:

- read presets from configMap with resources and buildResources for function
- add new tab `Resources`
- update tests and texts

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/9227

### CODE REVIEW

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ConfigurationTab.js [hunk 1]
```diff
@@ -1,45 +1,31 @@
...
-  backendModulesExist,
-} from 'components/Lambdas/helpers/misc';
 
 export default function ConfigurationTab({
   lambda,
   setBindingUsages = () => void 0,
-  backendModules = [],
+  apiGatewayExists = false,
+  eventingExists = false,
+  serviceCatalogExists = false,
```
Comment (aerfio) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/Configuration/ConfigurationTab.js:12]: wouldn't it be easier if we accepted `children` here and just inline those components in the parent? no playing with props this way

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/ResourceManagement/ResourceManagement/ResourceManagement.scss [hunk 2]
```diff
@@ -1,12 +1,22 @@
 .lambda-resources-management {
+  box-shadow: none !important;
```
Comment (aerfio) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/ResourceManagement/ResourceManagement/ResourceManagement.scss:2]: Is important really needed?

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/ResourceManagement/ResourceManagement/ResourceManagement.scss [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
+
+  > * {
```
Comment (aerfio) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/ResourceManagement/ResourceManagement/ResourceManagement.scss:4]: be more specific than `*`

**Review by aerfio — CHANGES_REQUESTED** [review]

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/ResourceManagement/ResourceManagement/shared.js [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+  }
+  return customPreset;
+}
+
+export function isCustomPreset(preset) {
+  return preset === customPreset;
+}
+
+export function updateResourcesValidationSchema() {
+  Object.assign(schema, prepareSchema());
```
Comment (aerfio) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/ResourceManagement/ResourceManagement/shared.js:337]: no return here?

**Review by aerfio — CHANGES_REQUESTED** [review]

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/ResourceManagement/ResourceManagement/ResourceManagement.js [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+  }
+
+  return (
+    <Panel className="fd-has-margin-m lambda-resources-management">
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Panel.Header className="fd-has-padding-xs">
+          <Panel.Head title={RESOURCES_MANAGEMENT_PANEL.TITLE} />
+          <Panel.Actions>
+            {renderCancelButton()}
+            {renderConfirmButton()}
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/ResourceManagement/ResourceManagement/ResourceManagement.js:215]: I made it some 3 months ago and I unlined for readability.

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/ResourceManagement/ResourceManagement/shared.js (hunk 4 above)
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/ResourceManagement/ResourceManagement/shared.js:337]: Nope. It operates on reference.

**Comment by aerfio** [comment]
No longer relevant

**Comment by aerfio** [comment]
Whoops wrong chrome tab 😛

File: core-ui/src/components/Lambdas/constants.js [hunk 6]
```diff
@@ -427,6 +454,16 @@ export const RESOURCES_MANAGEMENT_PANEL = {
     },
   },
   RESOURCES: {
+    TYPES: {
+      FUNCTION: {
+        TITLE: 'Runtime Profile',
+        DESCRIPTION: `Choose one of predefined Function's resources values or compose your own values ​​by selecting the "Custom" option`,
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:460]:
```suggestion
        DESCRIPTION: `Choose one of the predefined values for Function's resources or set your own values ​​by selecting the "Custom" option.`,
```

File: core-ui/src/components/Lambdas/constants.js [hunk 7]
```diff
@@ -427,6 +454,16 @@ export const RESOURCES_MANAGEMENT_PANEL = {
...
+      },
+      BUILD_JOB: {
+        TITLE: 'Build Job Profile',
+        DESCRIPTION: `Choose one of predefined build Job's resources values or compose your own values ​​by selecting the "Custom" option`,
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:464]:
```suggestion
        DESCRIPTION: `Choose one of the predefined values for build Job's resources or set your own values ​​by selecting the "Custom" option.`,
```

**Review by aerfio — DISMISSED** [review]

**Review by kazydek — APPROVED** [review]

**Review by kazydek — APPROVED** [review]

**Review by aerfio — APPROVED** [review]

**Pull request merged**